external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaUtils.java

```diff
@@ -213,7 +213,11 @@ public static ByteBufferMessageSet fetchMessages(KafkaConfig config, SimpleConsu
                 throw new TopicOffsetOutOfRangeException(msg);
             } else {
                 String message = "Error fetching data from [" + partition + "] for topic [" + topic + "]: [" + error + "]";
-                LOG.error(message);
+                if (error.equals(KafkaError.NOT_LEADER_FOR_PARTITION)) {
```

Comment:
It should get caught here https://github.com/apache/storm/blob/master/external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaSpout.java#L142 or here https://github.com/apache/storm/blob/master/external/storm-kafka/src/jvm/org/apache/storm/kafka/trident/TridentKafkaEmitter.java#L78